dockstore-common/src/main/java/io/dockstore/common/yaml/AuthorNameOrOrcid.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dockstore.common.yaml;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Target({ ElementType.TYPE, ElementType.ANNOTATION_TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = AuthorNameOrOrcidValidator.class)
+public @interface AuthorNameOrOrcid {
```

Comment:
Javadoc this, potentially more descriptive name. 
This looks like a message class storing errors having to do with authors or orcid ids?

Reply:
Also for other reviewers https://stackoverflow.com/questions/918393/whats-the-difference-between-interface-and-interface-in-java

Reply:
> Javadoc this, potentially more descriptive name. This looks like a message class storing errors having to do with authors or orcid ids?

This defines the annotation for the constraint that checks the author name+orcid.  Yeah, some javadoc and maybe a name change to `AuthorHasNameOrOrcid` would help a lot...